### ISSUE TITLE
Retry-After field in 503 replies not handled

### ISSUE THREAD

**Issue by PavloTV** [onsip/SIP.js#472]
When SIP.js trying to register on FreeSWITCH and sip profiles are not ready, server reply with 503 Maximum Calls In Progress with Retry-After: 300 field. After this answer client stops any registration attempts (i am waiting about 30mins).

> ### REGISTER sip:example.com SIP/2.0
> Via: SIP/2.0/WSS uk4ud1kl6e6g.invalid;branch=z9hG4bK2437045
> Max-Forwards: 70
> To: <sip:769@example.com>
> From: <sip:769@example.com>;tag=3svruftkku
> Call-ID: jcckccr3t3n5vkiui4ncbg
> CSeq: 87 REGISTER
> Authorization: Digest algorithm=MD5, username="769", realm="example.com", nonce="31bd99c4-c02f-4377-981c-35c66339f1c9", uri="sip:example.com", response="c3082958e5238b8008aa356e031ff1ec", qop=auth, cnonce="9d96uef9p7a2", nc=00000001
> Contact: <sip:1oa2i69j@uk4ud1kl6e6g.invalid;transport=ws>;reg-id=1;+sip.instance="<urn:uuid:75201fa0-920b-4f70-8b0c-c0240f91807b>";expires=300
> Allow: ACK,CANCEL,INVITE,MESSAGE,BYE,OPTIONS,INFO,NOTIFY,REFER
> Supported: path, gruu, outbound
> User-Agent: SIP.js/0.7.8
> Content-Length: 0

> ### SIP/2.0 503 Maximum Calls In Progress
> Via: SIP/2.0/WSS uk4ud1kl6e6g.invalid;branch=z9hG4bK2437045;received=46.123.45.78;rport=49848
> From: <sip:769@>;tag=3svruftkku
> To: <sip:769@example.com>;tag=QBZNN77BSKayc
> Call-ID: jcckccr3t3n5vkiui4ncbg
> CSeq: 87 REGISTER
> ### Retry-After: 300
> User-Agent: FreeSWITCH-mod_sofia/1.4.26-37~64bit
> Allow: INVITE, ACK, BYE, CANCEL, OPTIONS, MESSAGE, INFO, UPDATE, REGISTER, REFER, NOTIFY, PUBLISH, SUBSCRIBE
> Supported: path, replaces
> Content-Length: 0

According to https://tools.ietf.org/html/rfc3261, Retry-After field can be used with 404,413,480,486,500,503,600,603 replies.

**Comment by c960657** [comment]
Note that FreeSWITCH does not allow you to configure the value of `Retry-After`, so if SIP.js decides to use this value, please make it possible to override the value returned by the server.

**Comment by john-e-riordan** [comment]
Technically, I don't believe this is a "bug" because I haven't found anything that says Retry-After must be handled. That said, it would be nice to be able to handle it. I took a stab in #799.

I didn't have the API deal with doing the actual retry. Instead I provided a means to get the Retry-After value when a 500 or 503 is rejects a registration attempt via new property - `Registerer.retryAfter` - which is available on `stateChange` to `RegistererState.Unregistered`. Thus the developer can choose what, if anything, to do with it. Coding up the retry should be straight forward and as @c960657 notes should be application dependent.
